test/parallel/test-buffer-copy.js

```diff
@@ -144,3 +144,36 @@ assert.strictEqual(b.copy(c, 512, 0, 10), 0);
     assert.strictEqual(c[i], e[i]);
   }
 }
+
+
+// Validate arg type checking.
+// Refs: https://github.com/nodejs/node/issues/23668
+{
+  common.expectsError(
+    () => b.copy(c, '1'),
+    {
+      code: 'ERR_INVALID_ARG_TYPE',
+      type: TypeError,
+      message: 'The "targetStart" argument must be of type ' +
+        'Number. Received type string'
```

Comment:
Is there a reason for the inconsistent capitalization between 'string' and 'Number'?

Reply:
(that above is a super-minor non-blocking nit, probably applies to lots of other errors in our code base, can be cleaned up later, but if you're feeling in a perfectionist mood...)

Reply:
That's the dark magic of `ERR_INVALID_ARG_TYPE('targetStart', ['Number'], targetStart)`. I'm assuming it does `.name` to the types, and `typeof val` to the val...

https://github.com/nodejs/node/blob/acb363ffc40a0a76f7314385309a6d4ed78ef354/lib/internal/errors.js#L674-L699